From c20a515b6d88b56973b628c35a6dc08c8e666084 Mon Sep 17 00:00:00 2001
From: Eugene Syromyatnikov <evgsyr@gmail.com>
Date: Wed, 20 Jun 2018 18:57:07 +0200
Subject: [PATCH] xlat: remove non-Linux flags from open_mode_flags

* xlat/open_mode_flags.in (O_PRIV, FNDELAY, FAPPEND, FMARK, FDEFER,
FSHLOCK, FEXLOCK, FCREAT, FTRUNC, FEXCL, FNBIO, FSYNC, FNOCTTY,
O_SHLOCK, O_EXLOCK): Remove flags that are not present in Linux.

Co-Authored-by: Dmitry V. Levin <ldv@altlinux.org>
---
 xlat/open_mode_flags.in | 15 ---------------
 1 file changed, 15 deletions(-)

Index: strace-4.23/xlat/open_mode_flags.in
===================================================================
--- strace-4.23.orig/xlat/open_mode_flags.in	2018-02-07 00:46:18.000000000 +0100
+++ strace-4.23/xlat/open_mode_flags.in	2018-08-09 10:17:01.967562204 +0200
@@ -13,7 +13,6 @@
 #if defined(O_NDELAY) && (O_NDELAY != O_NONBLOCK)
 O_NDELAY
 #endif
-O_PRIV
 O_DIRECT
 O_LARGEFILE
 O_NOFOLLOW
@@ -24,18 +23,4 @@
 O_TMPFILE
 __O_TMPFILE
 O_DIRECTORY
-FNDELAY
-FAPPEND
-FMARK
-FDEFER
 FASYNC
-FSHLOCK
-FEXLOCK
-FCREAT
-FTRUNC
-FEXCL
-FNBIO
-FSYNC
-FNOCTTY
-O_SHLOCK
-O_EXLOCK
Index: strace-4.23/xlat/open_mode_flags.h
===================================================================
--- strace-4.23.orig/xlat/open_mode_flags.h	2018-06-14 20:03:43.000000000 +0200
+++ strace-4.23/xlat/open_mode_flags.h	2018-08-09 10:18:40.442672165 +0200
@@ -50,9 +50,6 @@
   XLAT(O_NDELAY),
 #endif
 #endif
-#if defined(O_PRIV) || (defined(HAVE_DECL_O_PRIV) && HAVE_DECL_O_PRIV)
-  XLAT(O_PRIV),
-#endif
 #if defined(O_DIRECT) || (defined(HAVE_DECL_O_DIRECT) && HAVE_DECL_O_DIRECT)
   XLAT(O_DIRECT),
 #endif
@@ -81,51 +78,9 @@
 #if defined(O_DIRECTORY) || (defined(HAVE_DECL_O_DIRECTORY) && HAVE_DECL_O_DIRECTORY)
   XLAT(O_DIRECTORY),
 #endif
-#if defined(FNDELAY) || (defined(HAVE_DECL_FNDELAY) && HAVE_DECL_FNDELAY)
-  XLAT(FNDELAY),
-#endif
-#if defined(FAPPEND) || (defined(HAVE_DECL_FAPPEND) && HAVE_DECL_FAPPEND)
-  XLAT(FAPPEND),
-#endif
-#if defined(FMARK) || (defined(HAVE_DECL_FMARK) && HAVE_DECL_FMARK)
-  XLAT(FMARK),
-#endif
-#if defined(FDEFER) || (defined(HAVE_DECL_FDEFER) && HAVE_DECL_FDEFER)
-  XLAT(FDEFER),
-#endif
 #if defined(FASYNC) || (defined(HAVE_DECL_FASYNC) && HAVE_DECL_FASYNC)
   XLAT(FASYNC),
 #endif
-#if defined(FSHLOCK) || (defined(HAVE_DECL_FSHLOCK) && HAVE_DECL_FSHLOCK)
-  XLAT(FSHLOCK),
-#endif
-#if defined(FEXLOCK) || (defined(HAVE_DECL_FEXLOCK) && HAVE_DECL_FEXLOCK)
-  XLAT(FEXLOCK),
-#endif
-#if defined(FCREAT) || (defined(HAVE_DECL_FCREAT) && HAVE_DECL_FCREAT)
-  XLAT(FCREAT),
-#endif
-#if defined(FTRUNC) || (defined(HAVE_DECL_FTRUNC) && HAVE_DECL_FTRUNC)
-  XLAT(FTRUNC),
-#endif
-#if defined(FEXCL) || (defined(HAVE_DECL_FEXCL) && HAVE_DECL_FEXCL)
-  XLAT(FEXCL),
-#endif
-#if defined(FNBIO) || (defined(HAVE_DECL_FNBIO) && HAVE_DECL_FNBIO)
-  XLAT(FNBIO),
-#endif
-#if defined(FSYNC) || (defined(HAVE_DECL_FSYNC) && HAVE_DECL_FSYNC)
-  XLAT(FSYNC),
-#endif
-#if defined(FNOCTTY) || (defined(HAVE_DECL_FNOCTTY) && HAVE_DECL_FNOCTTY)
-  XLAT(FNOCTTY),
-#endif
-#if defined(O_SHLOCK) || (defined(HAVE_DECL_O_SHLOCK) && HAVE_DECL_O_SHLOCK)
-  XLAT(O_SHLOCK),
-#endif
-#if defined(O_EXLOCK) || (defined(HAVE_DECL_O_EXLOCK) && HAVE_DECL_O_EXLOCK)
-  XLAT(O_EXLOCK),
-#endif
  XLAT_END
 };
 
